### PR TITLE
React Dev Docs Final Cleanup/Fixes

### PR DESCRIPTION
**Pull request by joebochill**
Fixes BLUI-3552.

#### Changes proposed in this Pull Request:
- Clean up all logs and warnings
- Fix broken props (animationDuration)
- Fix Number picker for decimal values (opacity)
- Refactor routing to support tab preservation 
- Fix styling of tabs so they stay sticky on overscroll

#### To Test:
- EVERYTHING

### CODE REVIEW

File: components/src/core/AppBar/AppBar.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ const useUtilityClasses = (ownerState: AppBarProps): Record<AppBarClassKey, stri
 const Root = styled(MuiAppBar, {
     name: 'app-bar',
     slot: 'root',
-    shouldForwardProp: (prop) => prop !== 'backgroundImage',
+    shouldForwardProp: (prop) => !['backgroundImage', 'animationDuration'].includes(prop.toString()),
```
Comment (emclaug2) [inline components/src/core/AppBar/AppBar.tsx:29]: If you're making changes in the src folder, we're going to need a changelog entry description what you've changed.   Same applies for EmptyState change.

Comment (joebochill) [reply components/src/core/AppBar/AppBar.tsx:29]: Done

**Comment by jeffvg** [comment]
couple more not sure why this happened

shared props links in API docs table
image

drawer-rail-item/api-docs
sharedProps link missing route item /drawer
https://blui-react-docs--pr623-feature-3552-final-f-mn1pd67j.web.app/components/drawer-rail-item/api-docs#shared-props

should be - 
https://blui-react-docs--pr623-feature-3552-final-f-mn1pd67j.web.app/components/drawer/drawer-rail-item/api-docs#shared-props

drawer-body/api-docs
https://blui-react-docs--pr623-feature-3552-final-f-mn1pd67j.web.app/components/drawer-body/api-docs#shared-props

should be - 
https://blui-react-docs--pr623-feature-3552-final-f-mn1pd67j.web.app/components/drawer/drawer-body/api-docs#shared-props

drawer/api-docs
sharedProps link not able to navigate down to location in same page
https://blui-react-docs--pr623-feature-3552-final-f-mn1pd67j.web.app/components/drawer/drawer/api-docs#shared-props

**Comment by joebochill** [comment]
I fixed the links so they should work now. I would like to circle back and make it so that the drawer subcomponent routes aren't all prefixed by '/drawer/XXX', but I don't have the time to go in and make the routing changes now.

**Review by jeffvg — APPROVED** [review]

**Review by emclaug2 — APPROVED** [review]

**Pull request merged**